x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form_indicator_section.tsx

```diff
@@ -8,17 +8,38 @@
 import { EuiFormRow, EuiPanel, EuiSelect, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { CreateSLOInput } from '@kbn/slo-schema';
-import React from 'react';
-import { Controller, useFormContext } from 'react-hook-form';
+import React, { useEffect } from 'react';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+import { isObject } from 'lodash';
 
 import { SLI_OPTIONS } from '../constants';
 import { ApmAvailabilityIndicatorTypeForm } from './apm_availability/apm_availability_indicator_type_form';
 import { ApmLatencyIndicatorTypeForm } from './apm_latency/apm_latency_indicator_type_form';
 import { CustomKqlIndicatorTypeForm } from './custom_kql/custom_kql_indicator_type_form';
+import {
+  CustomMetricIndicatorTypeForm,
+  NEW_CUSTOM_METRIC,
+} from './custom_metric/custom_metric_type_form';
 import { maxWidth } from './slo_edit_form';
 
 export function SloEditFormIndicatorSection() {
-  const { control, watch } = useFormContext<CreateSLOInput>();
+  const { control, watch, setValue } = useFormContext<CreateSLOInput>();
+
+  const indicator = useWatch({ control, name: 'indicator.type' });
+
+  // Set the defaults for the form when switching between different indicators
+  useEffect(() => {
+    if (indicator === 'sli.metric.custom') {
+      setValue('indicator.params.good.metrics', [NEW_CUSTOM_METRIC]);
+      setValue('indicator.params.good.equation', 'A');
+      setValue('indicator.params.total.metrics', [NEW_CUSTOM_METRIC]);
+      setValue('indicator.params.total.equation', 'A');
+    }
```

Comment:
Yes... if you started on "Custom metric" then switched to "Custom KQL" then back, you would get a fresh form. The reason being is that `indicator.params.good` (and `total`) is a different shaped object on both on `Custom KQL`  (`string`) and `Custom Metric` (`object`). If we wanted to keep the state when switching between indicators, I would need to store `indicator.params.good` in something like `indicator.params.goodMetric`. 

This begs the question, "is it reasonable to expect the form to retain state when switching the indicator types?"

Reply:
I understand the reason but I'm wondering in the case of the **edit** flow (i'm gonna test this AM so I'll check that), if it does not erase the good and total metrics/equation when it loads the form?

> This begs the question, "is it reasonable to expect the form to retain state when switching the indicator types?"

I think for the sake of code simplicity and maintenance, that would be a fine approach by me.

Reply:
> I understand the reason but I'm wondering in the case of the edit flow (i'm gonna test this AM so I'll check that), if it does not erase the good and total metrics/equation when it loads the form?

tested and it does not, all good.

But I think we should maybe reset the form when we switch indicator type. It will simplify everything. wdyt?